### ISSUE TITLE
Cannot convert GetValue('a_uint8') to MI_Uint8A

### ISSUE THREAD

**Issue by Wayenyin** [Microsoft/omi-script-provider#64]
**Build**

- omi path:[\\redmond\wsscfs\OSTCData\Builds\omi\develop\1.4.0-37 ](\\redmond\wsscfs\OSTCData\Builds\omi\develop\1.4.0-37 )
- Python script:[\\redmond\wsscfs\OSTCData\Builds\omiscriptprovider\develop\1.1.1-35 ](\\redmond\wsscfs\OSTCData\Builds\omiscriptprovider\develop\1.1.1-35 )

**Reproduce steps**
1.	Write mof file:
class TestClass_AllDMTFArrays
{
        [key] uint64 v_Key;
        Boolean a_bool[];
        Uint8 a_uint8[];
};        
2.	Generate above test1.mof and add code like following in mi_main.py:
root@ omi64-cent7-01: /root/TestClass_AllDMTFArrays/; /opt/omi/bin/omigen_py -I /root/TestClass_AllDMTFArrays/ /root/TestClass_AllDMTFArrays/TestClass_AllDMTFArrays.mof TestClass_AllDMTFArrays

```
def TestClass_AllDMTFArrays_CreateInstance (
    context, nameSpace, className, instance):
    be = BookEnd ('TestClass_AllDMTFArrays_CreateInstance')
    createKey = instance.GetValue('v_Key').value
    createUint8A = MI_Uint8A(instance.GetValue('a_uint8'))
    instanceTemp = context.NewInstance ('TestClass_AllDMTFArrays')
    instanceTemp.SetValue('v_Key', MI_Uint64(createKey))
    instanceTemp.SetValue('a_bool', instance.GetValue('a_bool') )
    # instanceTemp.SetValue('a_uint8',createUint8A)             
    instanceStore.append(instanceTemp)
    context.PostResult (MI_RESULT_OK)
```
3.	Register your provider.
```
   /opt/omi/lib
   root@omi64-cent7-01  # /opt/omi/bin/omireg --Python TestClass_AllDMTFArrays
   Created /etc/opt/omi/conf/omiregister/root-cimv2/TestClass_AllDMTFArrays.reg
   /opt/omi/bin/omiserver: refreshed server
```
4.	Run ci
```
   /opt/omi/lib
  root@omi64-cent7-01  # /opt/omi/bin/omicli ci root/cimv2 { TestClass_AllDMTFArrays  v_Key 4 a_uint8 [ 1 ] }
  /opt/omi/bin/omicli: result: MI_RESULT_FAILED
  instance of OMI_Error
  {
      OwningEntity=OMI:CIMOM
      MessageID=OMI:MI_Result:1
      Message=A general error occurred, not covered by a more specific error code.
      MessageArguments={}
      PerceivedSeverity=7
      ProbableCause=0
      ProbableCauseDescription=Unknown
      CIMStatusCode=1
      OMI_Code=1
      OMI_Category=0
      OMI_Type=MI
      OMI_ErrorMessage=A general error occurred, not covered by a more specific error code.
   }
```

**NOTE:** If you try to set as following, it will work fine.	
- modify mi_main.py file:
```
  def TestClass_AllDMTFArrays_CreateInstance (
        context, nameSpace, className, instance):
        be = BookEnd ('TestClass_AllDMTFArrays_CreateInstance')
        createKey = instance.GetValue('v_Key').value
        instanceTemp = context.NewInstance ('TestClass_AllDMTFArrays')
        instanceTemp.SetValue('v_Key', MI_Uint64(createKey))
        instanceStore.append(instanceTemp)
        context.PostResult (MI_RESULT_OK)
```
- Run ci command, then run ei to check new instance
```
   root@omi64-cent7-01  # /opt/omi/bin/omicli ci root/cimv2 { TestClass_AllDMTFArrays  v_Key 4 a_bool [ true ] }

   /opt/omi/lib
   root@omi64-cent7-01  # /opt/omi/bin/omicli ei root/cimv2  TestClass_AllDMTFArrays
   instance of TestClass_AllDMTFArrays
   {
       [Key] v_Key=0
       a_bool={true, false, true, false}
   }
   instance of TestClass_AllDMTFArrays
   {
      [Key] v_Key=1
      a_bool={true, false, true, false}
   }
   instance of TestClass_AllDMTFArrays
   {
       [Key] v_Key=2
       a_bool={true, false, true, false}
   }
   instance of TestClass_AllDMTFArrays
   {
       [Key] v_Key=3
       a_bool={true, false, true, false}
   }
   instance of TestClass_AllDMTFArrays
   {
       [Key] v_Key=4
   }
```

**Comment by Wayenyin** [comment]
Following types also cannot convert:
-  Cannot convert GetValue('a_bool')  to  MI_BooleanA()
-  Cannot convert  GetValue('a_uint8') to  MI_Uint8A()
-  Cannot convert  GetValue('a_sint8')  to MI_Sint8A()
-  Cannot convert  GetValue('a_UINT16') to  MI_Uint16A()
-  Cannot convert  GetValue('a_sint16')  to MI_Sint16A()
-  Cannot convert  GetValue('a_uint32') to  MI_Uint32A()
-  Cannot convert  GetValue('a_sint32')  to MI_Sint32A()
-  Cannot convert  GetValue('a_uint64') to  MI_Uint64A()
-  Cannot convert  GetValue('a_sint64')  to MI_Sint64A()
-  Cannot convert  GetValue('a_rEal32') to  MI_Real32A()
-  Cannot convert  GetValue('a_REAL64')  to MI_Real64A()
-  Cannot convert  GetValue('a_char16') to  MI_Char16A()
-  Cannot convert  GetValue('a_string')  to MI_StringA()
-  Cannot convert  GetValue('a_DATETIME')  to MI_DatetimeA()

**Comment by alinbalutoiu** [comment]
I am just curious, have you tested all the types? Or you're just guessing that? If so, please **stop** making this kind of guesses.
To fix your issue, please change the code to:
```
def TestClass_AllDMTFArrays_CreateInstance (
    context, nameSpace, className, instance):
    be = BookEnd ('TestClass_AllDMTFArrays_CreateInstance')
    createKey = instance.GetValue('v_Key').value
    createUint8A = instance.GetValue('a_uint8')
    instanceTemp = context.NewInstance ('TestClass_AllDMTFArrays')
    instanceTemp.SetValue('v_Key', MI_Uint64(createKey))
    # instanceTemp.SetValue('a_bool', instance.GetValue('a_bool') )
    instanceTemp.SetValue('a_uint8',createUint8A)             
    instanceStore.append(instanceTemp)
    context.PostResult (MI_RESULT_OK)
```
Could you please try to make sense when you're writing the issue and paste the code snippet? You're talking about "Cannot convert GetValue('a_uint8') to MI_Uint8A", yet in the code you were setting a_bool which you were not even passing via create instance command.
You're mixing some stuff here. Try to avoid this kind of issues, otherwise it will make it harder to reproduce.

Anyway, I tried multiple values and they seem to work for me ('a_uint8', 'a_bool', 'a_uint16').
When you do:
```
createUint8A = instance.GetValue('a_uint8')
```
It already has the type MI_Uint8A and there is no need to add the cast to it like you're doing.
The short version would be:
```
instanceTemp.SetValue('a_uint8', instance.GetValue('a_uint8'))
```
For me seems to work without issue:
```
$ omicli ci root/cimv2 { TestClass_AllDMTFArrays  v_Key 4 a_uint8 [ 1 ] };
$ omicli ei root/cimv2 TestClass_AllDMTFArrays
instance of TestClass_AllDMTFArrays
{
    [Key] v_Key=4
    a_uint8={1}
}
```

Next time, please be more careful when you open an issue and try to paste the code to which you are referring to. Thanks!

**Comment by Wayenyin** [comment]
I didn't guessing for these kinds of cases, I wrote the wrong python code so get the error. I am so sorry for this, and I will be careful next time, thanks for your help.
